grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1141,4 +1142,46 @@ class ProductService {
         }
         return results
     }
+
+    def searchProductDtos(String[] terms) {
+        def query = """
+select p.id, p.name, p.product_code as productCode, p.cold_chain as coldChain, p.controlled_substance as controlledSubstance, p.hazardous_material as hazardousMaterial, p.reconditioned,
+ (select max(pc.color) from product_catalog_item pci left outer join product_catalog pc on pci.product_catalog_id = pc.id where pci.product_id = p.id group by pci.product_id) as productColor
+ from product p
+ left outer join product_supplier productsup1_ on p.id=productsup1_.product_id
+ left outer join party manufactur2_ on productsup1_.manufacturer_id=manufactur2_.id and (${terms.collect { "lower(manufactur2_.name) like '%${it}%'" }.join(" or ")})
```

Comment:
Is it possible to get better readability on the namings such as productsup1_, manufactur2_, supplier_a3_, and so on? It looks kinda odd (looks like it is copied from SQL loggs)

Reply:
Yeah I'm actually working on that right now.

Reply:
Here was my best attempt at making it more readable. I removed all of the aliases except the two party joins). 
```
    def searchProductDtos(String[] terms) {
        def query = """
            select 
            product.id, 
            product.name, 
            product.product_code as productCode, 
            product.cold_chain as coldChain, 
            product.controlled_substance as controlledSubstance, 
            product.hazardous_material as hazardousMaterial, 
            product.reconditioned,
            (
                select max(pc.color) 
                from product_catalog_item pci 
                left outer join product_catalog pc on pci.product_catalog_id = pc.id 
                where pci.product_id = product.id 
                group by pci.product_id
            ) as productColor
            from product
            left outer join product_supplier 
                on product.id = product_supplier.product_id
            left outer join party manufacturer 
                on product_supplier.manufacturer_id = manufacturer.id 
                and (${terms.collect { "lower(manufacturer.name) like '${it}%'" }.join(" or ")})
            left outer join party supplier 
                on product_supplier.supplier_id = supplier.id 
                and (${terms.collect { "lower(supplier.name) like '${it}%'" }.join(" or ")})
            left outer join inventory_item 
                on product.id = inventory_item.product_id 
                and (${terms.collect { "lower(inventory_item.lot_number) like '%${it}%'" }.join(" or ")})
            where product.active = 1 and (${terms.collect {"""
                lower(product.name) like '%${it}%' 
                or lower(product.product_code) like '${it}%' 
                or lower(product.description) like '%${it}%'
                or lower(product.upc) like '${it}%' 
                or lower(product.ndc) like '${it}%'
                or lower(product.unit_of_measure) like '${it}%' 
                or lower(product_supplier.name) like '%${it}%' 
                or lower(product_supplier.code) like '${it}%'
                or lower(product_supplier.product_code) like '${it}%' 
                or lower(product_supplier.manufacturer_code) like '${it}%'
                or lower(product_supplier.manufacturer_name) like '${it}%' 
                or lower(product_supplier.supplier_code) like '${it}%'
                or lower(product_supplier.supplier_name) like '${it}%'""" }.join(" or ")}
                or manufacturer.id is not null
                or supplier.id is not null
                or inventory_item.id is not null)
                group by product.id"""
        def results = dataService.executeQuery(query)
        return results.collect { new ProductSearchDto(it) }
    }
```